### PR TITLE
[SHLWAPI][SHLWAPI_APITEST][SDK] Implement SHPropertyBag_WritePOINTL etc.

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Implementing missing features...
JIRA issue: [CORE-9283](https://jira.reactos.org/browse/CORE-9283)

## Proposed changes

- Implement `SHPropertyBag_Delete`, `SHPropertyBag_WritePOINTL`, `SHPropertyBag_WritePOINTS`, and `SHPropertyBag_WriteRECTL` functions.
- `SHPropertyBag_WriteInt` is an alias to `SHPropertyBag_WriteLONG`.
- Modify `shlwapi.spec`.
- Modify `shlwapi_undoc.h`.
- Add `SHPropertyBag` testcase to `shlwapi_apitest.exe`.

## Comparison

WinXP:
WinXP
Successful.

Win2k3:
Win2k3
Successful.

Win10:
Win10
Successful.

ReactOS:
Ros
Successful.

### CODE REVIEW

File: dll/win32/shlwapi/ordinal.c
Comment (JoachimHenze) [inline dll/win32/shlwapi/ordinal.c:5328]: Why LPCOLESTR here instead of LPCWSTR?

File: sdk/include/reactos/shlwapi_undoc.h
Comment (JoachimHenze) [inline sdk/include/reactos/shlwapi_undoc.h:102]: Why LPCOLESTR here instead of LPCWSTR?

File: dll/win32/shlwapi/ordinal.c
Comment (katahiromz) [reply dll/win32/shlwapi/ordinal.c]: Fixed.

File: sdk/include/reactos/shlwapi_undoc.h
Comment (katahiromz) [reply sdk/include/reactos/shlwapi_undoc.h]: Fixed.

File: dll/win32/shlwapi/ordinal.c
Comment (learn-more) [inline dll/win32/shlwapi/ordinal.c:5341]: Do you have testcases for this?

Comment (katahiromz) [reply dll/win32/shlwapi/ordinal.c:5341]: No, I haven't.
I will add it in the future.

Comment (katahiromz) [reply dll/win32/shlwapi/ordinal.c:5341]: `modules/rostests/winetests/shlwapi/ordinal.c`

This will work in the future.

**Review by JoachimHenze — APPROVED** [review]
Approved. Covering the type compatibility of the current signature choice PCWSTR against OLESTR would be a nice thing to have, in case you would have any good ideas for tests in that context. I do recommend to wait for a 2nd approval (or 1 week if no other will come in) until merging.

**Comment by learn-more** [comment]
> Covering the type compatibility of the current signature choice PCWSTR against OLESTR would be a nice thing to have

What do you mean with this?

File: dll/win32/shlwapi/ordinal.c
Comment (HBelusca) [inline dll/win32/shlwapi/ordinal.c:5556]: Shouldn't this use the function that write "long" points? Does windows do the same, or does it implement it the same way as WritePOINTL, **but** uses WriteSHORT for storing the coordinates?

Comment (katahiromz) [reply dll/win32/shlwapi/ordinal.c:5556]: Windows does the same thing of this code.

Comment (katahiromz) [reply dll/win32/shlwapi/ordinal.c:5556]: The property bag is a named-property-set.

File: dll/win32/shlwapi/ordinal.c
Comment (learn-more) [inline dll/win32/shlwapi/ordinal.c:5608]: Top and left will remain in the property bag in this case?

Comment (katahiromz) [reply dll/win32/shlwapi/ordinal.c:5608]: Deleting either one is Okay.

File: dll/win32/shlwapi/ordinal.c
Comment (katahiromz) [reply dll/win32/shlwapi/ordinal.c:5341]: Added `SHPropertyBag` testcase.

File: dll/win32/shlwapi/ordinal.c
Comment (learn-more) [reply dll/win32/shlwapi/ordinal.c:5608]: But why delete just one value in the failure case, and not everything?

**Review by learn-more — APPROVED** [review]

File: dll/win32/shlwapi/ordinal.c
Comment (katahiromz) [reply dll/win32/shlwapi/ordinal.c:5608]: > But why delete just one value in the failure case, and not everything?

`SHPropertyBag` testcase did cover it.

**Comment by JoachimHenze** [comment]
> > Covering the type compatibility of the current signature choice PCWSTR against OLESTR would be a nice thing to have
> 
> What do you mean with this?

Well in the beginning Katayama intended to implemented some of the functions signatures using LPCOLESTR, and reviewers commented that LPCWSTR should be fine to be used here. I wonder whether that hope/claim could be proved by any test.

**Pull request merged**
